pkg/sqlite/load.go

```diff
@@ -1320,9 +1320,10 @@ func (s *sqlLoader) rmChannelEntry(tx *sql.Tx, csvName string) error {
 
 func getTailFromBundle(tx *sql.Tx, head string) (bundles []string, err error) {
 	getReplacesSkips := `SELECT replaces, skips FROM operatorbundle WHERE name=?`
-	isDefaultChannelHead := `SELECT head_operatorbundle_name FROM channel
-							INNER JOIN package ON channel.name = package.default_channel
-							WHERE channel.head_operatorbundle_name = ?`
+	isDefaultChannelHead := `SELECT head_operatorbundle_name FROM package, channel
+								WHERE channel.package_name == package.name
+								AND package.default_channel == channel.name
+								AND channel.head_operatorbundle_name = ?`
```

Comment:
see that the select with the inner join is wrong and is returning the head_operatorbundle_name when the bundle is head of a channel that is NOT the default channel of the package:


<img width="401" alt="Screen Shot 2021-09-16 at 16 36 19" src="https://user-images.githubusercontent.com/7708031/133642041-84391acb-4bf6-4102-8b8e-cac4be35aba2.png">
<img width="436" alt="Screen Shot 2021-09-16 at 16 36 04" src="https://user-images.githubusercontent.com/7708031/133642045-2e5e1628-0ebe-42eb-9763-b7f0c4271bb0.png">
<img width="676" alt="Screen Shot 2021-09-16 at 16 35 28" src="https://user-images.githubusercontent.com/7708031/133642050-32190d4f-d3a4-4691-b0f4-eb61421e048c.png">

Reply:
I verified this, the inner join was missing a compare on package name.
`SELECT head_operatorbundle_name FROM channel
    INNER JOIN package ON channel.name = package.default_channel
    AND channel.package_name = package.name
    WHERE channel.head_operatorbundle_name = ?`